system/Database/BasePreparedQuery.php

```diff
@@ -101,24 +102,74 @@ abstract public function _prepare(string $sql, array $options = []);
      * Takes a new set of data and runs it against the currently
      * prepared query. Upon success, will return a Results object.
      *
-     * @return ResultInterface
+     * @return bool|ResultInterface
+     *
+     * @throws Exception
```

Comment:
Why throws `Exception`?
If the query execution fails, it seems it should throw `DatabaseException`.

Reply:
Actually, it will throw different exceptions depending on the driver. For MySQL it will be `mysqli_sql_exception` for others `ErrorException`. There is no point in changing this and hiding the actual exception - at least I don't think it would be appropriate. I believe that the `DatabaseException` should be used for things that are coming from the framework side and are in our gesture to handle.

Reply:
We have changed the exception classes already in v4.3. 
See #6163 and https://github.com/codeigniter4/CodeIgniter4/blob/4.3/user_guide_src/source/changelogs/v4.3.0.rst#id1

When CI throws many kind of Exception, if we switch DBMS, our code will break when we want to catch the execption.
Or we must always catch `Exception`. But it is too wide. It is not good practice.

Anyway this PR is for `develop`. So it seems okay. But what about 4.3?